sim/battle.ts

```diff
@@ -761,7 +771,24 @@ export class Battle extends Dex.ModdedDex {
 				handlers.unshift({status: effect, callback, statusData: {}, end: null, thing: target});
 			}
 		}
+
+		let targetRelayVars = [];
+		if (target instanceof Array) {
+			if (relayVar instanceof Array) {
+				targetRelayVars = relayVar;
+			} else {
+				for (let i = 0; i < target.length; i++) targetRelayVars[i] = true;
+			}
+		}
 		for (const handler of handlers) {
+			if (handler.index !== undefined) {
+				if (!targetRelayVars[handler.index]) continue;
```

Comment:
Counter + Disguise isn't working because `targetRelayVars[handler.index]` in that case is 0, referring to the amount of damage dealt. However, I can't seem to get this code to run in that case and only that case, because everything I've tried breaks at least 5 other tests, including checking the event id and skipping that check the first time through the loop.